asab/metrics/influxdb.py

```diff
@@ -191,15 +191,45 @@ def metric_to_influxdb(metric_record, now):
 			values_lines.append(build_metric_line(field.get("tags").copy(), {"sum": field.get("values").get("sum")}))
 			values_lines.append(build_metric_line(field.get("tags").copy(), {"count": field.get("values").get("count")}))
 
-	else:
-		for field in fieldset:
-			# SKIP empty fields
-			if not field.get("values") or field.get("values") == {}:
-				continue
-			values_lines.append(build_metric_line(field.get("tags"), field.get("values")))
-
-	return ["{},{} {}\n".format(name, line, int(timestamp * 1e9)) for line in values_lines]
-
+    else:
+        for field in fieldset:
+            # SKIP empty fields
+            if not field.get("values") or field.get("values") == {}:
+                continue
+            values_lines.append(
+                build_metric_line(
+                    # validate tags
+                    validate_tags(field.get("tags")),
+                    validate_values(field.get("values")),
+                )
+            )
+
+    return [
+        "{},{} {}\n".format(name, line, int(timestamp * 1e9)) for line in values_lines
+    ]
+
+
+def validate_name(name: str):
+    return "".join(name.split()).replace(",", "")
```

Comment:
from: https://docs.influxdata.com/influxdb/v2.3/reference/syntax/line-protocol/#string
Line protocol supports special characters in [string elements](https://docs.influxdata.com/influxdb/v2.3/reference/syntax/line-protocol/#string). In the following contexts, it requires escaping certain characters with a backslash (`\`)

As I understand it, you should escape the character with a backslash rather than deleting (replacing by an empty string). Please test my hypothesis against InfluxDB - check whether it can parse such fields.

Reply:
Fixed and you were right. Special characters are now replaced by a backslash.